cardano-cli/src/Cardano/CLI/Shelley/Commands.hs

```diff
@@ -190,7 +190,7 @@ data TransactionCmd
   | TxCalculateMinFee
       TxBodyFile
       (Maybe NetworkId)
-      ProtocolParamsFile
```

Comment:
Please change this to a sum type.

Reply:
Can I ask for a justification for this suggestion?

Reply:
`Either` implies that `Left` is an error and in this case `Left` isn't an error. A sum type is more appropriate here and makes it more clear what's happening.

Reply:
> `Either` implies that `Left` is an error

I don't see that in the documentation of Either, to be frank.

I've seen more than one introduction into algebraic types that explains them as reducible to `(,)` and `Either`.

So, I'm inclned to see `Either` possessing the same generality as `Maybe`.

Now, on the clarity grounds I could agree actually. That would allow per-sum-element documentation of the intent.

Reply:
However, when I agree that clarity can be improved, take a look at the rest of `TransactionCmd`:

```
data TransactionCmd
  = TxBuildRaw
      AnyCardanoEra
      [TxIn]
      [TxOutAnyEra]
      (Maybe Value)
      -- ^ Multi-Asset value
      (Maybe SlotNo)
      -- ^ Transaction lower bound
      (Maybe SlotNo)
      -- ^ Transaction upper bound
      (Maybe Lovelace)
      -- ^ Tx fee
      [CertificateFile]
      [(StakeAddress, Lovelace)]
      TxMetadataJsonSchema
      [ScriptFile]
      -- ^ Auxillary scripts
      [MetadataFile]
      (Maybe UpdateProposalFile)
      TxBodyFile
  | TxSign TxBodyFile [WitnessSigningData] (Maybe NetworkId) TxFile
  | TxCreateWitness TxBodyFile WitnessSigningData (Maybe NetworkId) OutputFile
  | TxAssembleTxBodyWitness TxBodyFile [WitnessFile] OutputFile
  | TxSubmit AnyConsensusModeParams NetworkId FilePath
  | TxMintedPolicyId ScriptFile
  | TxCalculateMinFee
      TxBodyFile
      (Maybe NetworkId)
      (Either GenesisFile ProtocolParamsFile)
      TxInCount
      TxOutCount
      TxShelleyWitnessCount
      TxByronWitnessCount
  | TxGetTxId (Either TxBodyFile TxFile)
```

You already can see `TxGetTxId` use `Either`, in the exact same fashion.

Reply:
@Jimbo4350, for documentation purposes, I could duplicate the explanation for the semantics I provided in https://github.com/input-output-hk/cardano-node/pull/2327/files#diff-cc9b567f3acd813c81e105c074ebd783719c74364c88a78c3c22638322354f6dR542-R545

Would you prefer that?

Reply:
Actually, did that, as that's a good idea.

Reply:
In the `Data.Either` docs:
```
The Either type is sometimes used to represent a value which is either correct or an error; by convention, the Left constructor is used to hold an error value and the Right constructor is used to hold a correct value (mnemonic: "right" also means "correct").
```
Please make the change and when you do I will approve this PR.

Reply:
"Sometimes", I think means something that's different from "always", right?

Reply:
@Jimbo4350, the change you asked for has been made.

It would really help, in the future, to distinguish between _optional improvements_ and _hard requirements_.

*Especially* in the context, where the directly surrounding code does not follow the requested structure.